src/protocol/modulus_conversion/convert_shares.rs

```diff
@@ -108,6 +108,36 @@ impl ConvertShares {
     }
 }
 
+pub async fn convert_all_shares<F: Field>(
+    ctx: &ProtocolContext<'_, F>,
+    input: &[u64],
+) -> Result<Vec<Vec<Replicated<F>>>, BoxError> {
+    let converted_shares = try_join_all(zip(repeat(ctx), input).enumerate().map(
+        |(record_id, (ctx, row))| async move {
+            ConvertShares::new(XorShares {
+                num_bits: 64,
+                packed_bits: *row,
+            })
+            .execute(
+                ctx.narrow(&ModulusConversion(record_id.try_into().unwrap())),
+                RecordId::from(record_id),
+            )
+            .await
+        },
+    ))
+    .await?;
+
+    //Transpose the output to return bit-wise results
+    let cols = vec![Replicated::new(F::ZERO, F::ZERO); converted_shares.len()];
+    let mut output = vec![cols; converted_shares[0].len()];
```

Comment:
This length is that number you've specified on for 118, `num_bits`. Let's use the same param for this rather than this approach which will fail if there is no element zero.

BTW, *this* seems like the number of columns - the number of bits.

Reply:
Yes these are the columns in convert share output which need to become rows in the transpose